### DISCUSSION TITLE
Atom services and HVD

### DISCUSSION THREAD

**Discussion by VeronikaKusova** [INSPIRE-MIF/helpdesk#164] [General]
Dear colleagues,

At the INSPIRE conference there was mentioned Atom fees are not API. 

We extend our atom services by OpenSearch technology, which enables to find particular dataset using its name, bbox etc. Can we consider this type of extended Atom service as API? (From our point of view in this case it is API.)

Here are some examples of requests: 
*  request for dataset(s) in ETRS89 and covering the particulat bbox: https://atom.cuzk.cz/get.ashx?crs=ETRS89&bbox=13.717523321526826,49.23324042654766,13.936954,519437316,49.35697620594279

* request for the dataset with id CZ-00025712- CUZK_CPX_693863 
https://atom.cuzk.cz/get.ashx?crs=5514&spatial_dataset_identifier_code=CZ-00025712-CUZK_CPX_693863

Thank you in advance for your opinion.
Veronika

**Comment by armin11** [comment]
Dear Veronika,

in our opinion ATOM Feeds are typical APIs, cause they can be used programmatically to download (INSPIRE) datasets based on the given metadata and catalogue interfaces (CSW). We tested it with a free and open source python lib and integrated it in our Geoportal. Some further tests have to be done, but it works as expected:
https://inspire-gpkg-cache.readthedocs.io/en/latest/introduction.html#inspire-gpkg-cache

Some more infos and videos [only in german]:
https://www.geoportal.rlp.de/article/Download_von_Daten_zur_Offline-Nutzung/

Best regards ;-) ,

Armin

**Comment by fabiovinci** [comment]
Dear all,

in the [Implementing Act on High Value Datasets](http://data.europa.eu/eli/reg_impl/2023/138/oj) when the APIs are mentioned there is the following footnote:

> _Such as direct access download services based on Directive 2007/2/EC_.

And the definition of a "_direct access download services_" is:

![direct_access_ds](https://github.com/INSPIRE-MIF/helpdesk/assets/22212406/98c16cfc-558c-4f5a-859e-5c09cceabe72)

This means that this type of service shall "_allow the retrieval of Spatial Objects based upon a query_".

![direct_access_ds_operations](https://github.com/INSPIRE-MIF/helpdesk/assets/22212406/b5c34ca4-b53a-41ee-936e-864842f5554b)

According to these considerations, it seems that an ATOM cannot allow direct access to a single spatial object in a dataset.

**Comment by armin11** [comment]
"such as" 😉

**Reply by fabiovinci** [reply]
Indeed "Such as (INSPIRE) **direct access** download services" and not "Such as (INSPIRE) download services"😉

In any case, in all official presentations there is the following mapping:

![image](https://github.com/INSPIRE-MIF/helpdesk/assets/22212406/64346364-3997-4760-a7a5-8761967a6c89)

**Comment by VeronikaKusova** [comment]
Dear all,
Thank you for your comments. I was thinking about it and I understand the table from the official presentation, but I also agree with Armin’s statement, that “such as” does not exclude Atom services. 
From my point of view it does not make sense in case of raster data. The WCS services provides same data as Atom services. I mean in both cases you get “a picture”. 
What do you think?

Veronika

PS: And I would like to wish you a happy new year! :sparkles:
